### PR TITLE
Allow limiting number of concurrently creating resources

### PR DESCRIPTION
**Pull request by gluke77**
#18

### CODE REVIEW

**Comment by nebril** [comment]
How exactly is it supposed to work? I may be missing something, but it seems like this won't stop createResource function from creating resources above desired concurrency level.

**Comment by nebril** [comment]
LGTM.

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -228,7 +234,8 @@ func Create(depGraph DependencyGraph) {
 	//TODO Make sure every KO gets created eventually
 }
 
-//Kosaraju's algorithm implementation https://en.wikipedia.org/wiki/Kosaraju%27s_algorithm
+//DetectCycles implements Kosaraju's algorithm https://en.wikipedia.org/wiki/Kosaraju%27s_algorithm
+//for dedecting cycles in graph.
```
Comment (nebril) [inline scheduler/scheduler.go]: s/dedecting/detecting
(although Kosaraju's algorithm isn't really a cycle detection algorithm at it's core).

**Comment by gluke77** [comment]
wait for me

**Pull request merged**